### PR TITLE
Add BuildStatusServices

### PR DESCRIPTION
**Pull request by mrfyda**
Includes support for the following operations:
* GET the build status for a commit
* POST a build status for a commit
* PUT a build status for a commit

https://confluence.atlassian.com/bitbucket/buildstatus-resource-779295267.html

### CODE REVIEW

File: src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala [hunk 1]
```diff
@@ -50,14 +52,13 @@ class BitbucketClient(key: String, secretKey: String, token: String, secretToken
...
+   * Does an API request
    */
-  private def post[D,T](request: Request[T], values: D)(implicit reader: Reads[T], writer: Writeable[D], contentType: ContentTypeOf[D]): RequestResponse[T] = withClientRequest { client =>
-
+  private def performRequest[D, T](method: String, request: Request[T], values: D)(implicit reader: Reads[T], writer: Writeable[D], contentType: ContentTypeOf[D]): RequestResponse[T] = withClientRequest { client =>
     val jpromise = client.url(request.url)
       .sign(requestSigner)
       .withFollowRedirects(follow = true)
-      .post(values)
+      .withMethod(method).withBody(values).execute()
```
Comment (rtfpessoa) [inline src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala]: what is the reason to start using the methods by their string name?

File: build.sbt [hunk 2]
```diff
@@ -14,7 +14,8 @@ resolvers += "Typesafe maven repository" at "http://repo.typesafe.com/typesafe/m
 
 libraryDependencies ++= Seq(
   jodaTime,
-  playWS
+  playWS,
+  scalatest
```
Comment (rtfpessoa) [inline build.sbt]: is this being used?

File: src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala (hunk 1 above)
Comment (mrfyda) [inline src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala]: to avoid duplicating the method

File: build.sbt (hunk 2 above)
Comment (mrfyda) [inline build.sbt]: no

Comment (rtfpessoa) [inline build.sbt]: then why is it here?

File: src/main/scala/com/codacy/client/bitbucket/service/BuildStatusServices.scala [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import com.codacy.client.bitbucket.client.{BitbucketClient, Request, RequestResponse}
+import play.api.libs.json.Json
+
+class BuildStatusServices(client: BitbucketClient) {
+
+  /*
+   * Gets a commit build status
+   *
+   */
+  def getBuildStatus(owner: String, repository: String, commit: String, key: String): RequestResponse[BuildStatus] = {
```
Comment (pedrorijo91) [inline src/main/scala/com/codacy/client/bitbucket/service/BuildStatusServices.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://www.codacy.com/app/Codacy/bitbucket-scala-client/file/1807008538/issues/source?bid=2474672&fileBranchId=3073963#l13)

File: src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala [hunk 4]
```diff
@@ -50,14 +52,13 @@ class BitbucketClient(key: String, secretKey: String, token: String, secretToken
   }
 
   /*
-   * Does an API post
+   * Does an API request
    */
-  private def post[D,T](request: Request[T], values: D)(implicit reader: Reads[T], writer: Writeable[D], contentType: ContentTypeOf[D]): RequestResponse[T] = withClientRequest { client =>
-
+  private def performRequest[D, T](method: String, request: Request[T], values: D)(implicit reader: Reads[T], writer: Writeable[D], contentType: ContentTypeOf[D]): RequestResponse[T] = withClientRequest { client =>
```
Comment (pedrorijo91) [inline src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala]: Codacy Issue found: [performRequest should have a smaller scope by using private[this]](https://www.codacy.com/app/Codacy/bitbucket-scala-client/file/1807008534/issues/source?bid=2474672&fileBranchId=3073963#l57)

File: src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala [hunk 5]
```diff
@@ -1,12 +1,14 @@
 package com.codacy.client.bitbucket.client
 
+import java.util.concurrent.{SynchronousQueue, ThreadPoolExecutor, TimeUnit}
+
 import com.codacy.client.bitbucket.util.HTTPStatusCodes
```
Comment (pedrorijo91) [inline src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala]: Codacy Issue found: [Imports should be sorted alphabetically](https://www.codacy.com/app/Codacy/bitbucket-scala-client/file/1807008534/issues/source?bid=2474672&fileBranchId=3073963#l5)

**Pull request merged**